reward/staking_manager.go

```diff
@@ -40,65 +41,89 @@ type blockChain interface {
 }
 
 type StakingManager struct {
-	ac           *addressBookConnector
-	sic          *stakingInfoCache
-	gh           governanceHelper
-	bc           blockChain
-	chainHeadCh  chan blockchain.ChainHeadEvent
-	chainHeadSub event.Subscription
-	isActivated  bool
+	addressBookConnector *addressBookConnector
+	stakingInfoCache     *stakingInfoCache
+	dbManager            database.DBManager // to store staking info into db
```

Comment:
`StakingManager` needs the DB interface like below. 
So why don't you define the interface and use it.
```
type stakingInfoDB interface {
	ReadStakingInfo(blockNum uint64) (interface{}, error)
	WriteStakingInfo(stakingInfo interface{}) error
}
```

```suggestion
	dbManager            stakingInfoDB
```

Reply:
I reflected this comment. Thank you for reviewing :)